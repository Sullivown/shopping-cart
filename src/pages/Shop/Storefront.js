import React, { useEffect } from 'react';
import updateDocumentTitle from '../../helpers/updateDocumentTitle';
import ProductPreview from '../../components/ProductPreview';

function Storefront(props) {
	useEffect(() => {
		updateDocumentTitle(props.title);
	});

	const productElements = props.products.map((product) => (
		<ProductPreview product={product} />
	));

	return (
		<div className='storefront'>
			<div className='page-title-div'>
				<h2>Shop</h2>
				<span>Displaying {props.products.length} items</span>
			</div>

			<div className='product-preview-container'>{productElements}</div>
		</div>
	);
}

export default Storefront;
