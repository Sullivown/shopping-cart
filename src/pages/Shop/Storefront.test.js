import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';

import Storefront from './Storefront';

const mockProducts = [
	{
		id: 1,
		title: 'Product 1',
		price: 2.99,
		image: jest.mock('../../assets/images/product-images/1.png'),
	},
	{
		id: 2,
		title: 'Product 2',
		price: 10,
		image: jest.mock('../../assets/images/product-images/2.png'),
	},
	{
		id: 3,
		title: 'Product 3',
		price: 5.99,
		image: jest.mock('../../assets/images/product-images/3.png'),
	},
];

describe('Storefront component testing', () => {
	const addToCartMock = jest.fn();
	it('renders', () => {
		const { container } = render(
			<Router>
				<Storefront
					products={mockProducts}
					title='Shop'
					addToCart={addToCartMock}
				/>
			</Router>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders 3 products', () => {
		render(
			<Router>
				<Storefront
					products={mockProducts}
					title='Shop'
					addToCart={addToCartMock}
				/>
			</Router>
		);

		const productList = screen.getByTestId('product-preview-container');
		expect(productList.childNodes.length).toBe(3);
	});

	it('renders correct number of products (3)', () => {
		render(
			<Router>
				<Storefront
					products={mockProducts}
					title='Shop'
					addToCart={addToCartMock}
				/>
			</Router>
		);

		const totalItems = screen.getByTestId('total-items');
		expect(totalItems.textContent).toBe('Displaying 3 items');
	});
});
