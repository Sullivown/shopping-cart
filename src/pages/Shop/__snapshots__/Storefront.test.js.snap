// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storefront component testing renders 1`] = `
<div>
  <div
    class="page-content storefront"
  >
    <div
      class="page-title-div"
    >
      <h2>
        Shop
      </h2>
      <span
        data-testid="total-items"
      >
        Displaying 
        3
         items
      </span>
    </div>
    <div
      class="product-preview-container"
      data-testid="product-preview-container"
    >
      <section
        class="product-preview"
      >
        <img
          alt="Product 1"
          src="[object Object]"
        />
        <h3>
          Product 1
        </h3>
        <p>
          £2.99
        </p>
        <button
          type="button"
        >
          View Item
        </button>
        <button>
          Add to Cart
        </button>
      </section>
      <section
        class="product-preview"
      >
        <img
          alt="Product 2"
          src="[object Object]"
        />
        <h3>
          Product 2
        </h3>
        <p>
          £10.00
        </p>
        <button
          type="button"
        >
          View Item
        </button>
        <button>
          Add to Cart
        </button>
      </section>
      <section
        class="product-preview"
      >
        <img
          alt="Product 3"
          src="[object Object]"
        />
        <h3>
          Product 3
        </h3>
        <p>
          £5.99
        </p>
        <button
          type="button"
        >
          View Item
        </button>
        <button>
          Add to Cart
        </button>
      </section>
    </div>
  </div>
</div>
`;
